### PR TITLE
Suggested updates from ROS

### PR DESCRIPTION
**Pull request by kraenhansen**
I suggest moving some updates to this template project from https://github.com/realm/ros.

More specifically:
- Remove the use of gulp
- Upgrade `typescript` and `ts-node`
- 2 space indention in the config files.
- Typescript targeting es6 and using a seperate dist folder for transpiled sources.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+    "target": "es6",
     "module": "commonjs",
     "moduleResolution": "node",
     "noImplicitAny": false,
     "removeComments": true,
     "preserveConstEnums": true,
     "sourceMap": true,
-    "forceConsistentCasingInFileNames": true
+    "forceConsistentCasingInFileNames": true,
+    "outDir": "dist",
```
Comment (mbalex99) [inline tsconfig.json:11]: can you add `"sourceRoot": "./src"`

File: package.json [hunk 2]
```diff
@@ -14,30 +14,22 @@
...
-    "test": "npm run build && ./node_modules/.bin/gulp --gulpfile gulpfile.js test",
-    "build": "./node_modules/.bin/gulp --gulpfile gulpfile.js build",
-    "watch": "./node_modules/.bin/gulp --gulpfile gulpfile.js watch",
-    "clean": "./node_modules/.bin/gulp --gulpfile gulpfile.js clean",
-    "lint": "./node_modules/.bin/tslint -c tslint.json 'src/**/*.ts'"
-  },
-  "dependencies": {
+    "test": "mocha --opts src/tests/mocha.opts",
+    "build": "rm -rf dist; tsc",
+    "watch": "tsc --watch",
```
Comment (mbalex99) [inline package.json]:
```javascript
    "build": "npm run clean && ./node_modules/.bin/tsc",
    "watch": "./node_modules/.bin/tsc --watch",
    "clean": "rm -rf dist",
    "lint": "./node_modules/.bin/tslint -c tslint.json 'src/**/*.ts'"
```

**Review by mbalex99 — COMMENTED** [review]
Good changes. I suggest referencing local dependencies.

File: package.json [hunk 3]
```diff
@@ -14,30 +14,22 @@
...
-    "gulp-mocha": "^4.3.1",
-    "gulp-sourcemaps": "^2.6.0",
-    "gulp-tslint": "^8.1.1",
-    "gulp-typescript": "^3.2.0",
-    "gulp-util": "^3.0.8",
     "mocha": "^3.4.2",
-    "ts-node": "^3.2.0",
+    "ts-node": "^3.3.0",
     "tslint": "^5.5.0",
-    "typescript": "^2.4.1"
```
Comment (dunstontc) [inline package.json:40]: 🤘

File: package.json (hunk 2 above)
Comment (kraenhansen) [reply package.json]: Makes sense to use the `clean` script, but there is really no need for `./node_modules/.bin/` in scripts.

https://docs.npmjs.com/misc/scripts#path

> If you depend on modules that define executable scripts, like test suites, then those executables will be added to the PATH for executing the scripts. So, if your package.json has this:
> 
>     { "name" : "foo"
>     , "dependencies" : { "bar" : "0.1.x" }
>     , "scripts": { "start" : "bar ./test" } }
> then you could run npm start to execute the bar script, which is exported into the node_modules/.bin directory on npm install.

**Comment by kraenhansen** [comment]
@mbalex99 - BTW. This project has the same issue with tests being ignored by the tsconfig.json as I created a PR for on the ROS repo.

**Pull request merged**